build.assets/Dockerfile-multiarch

```diff
@@ -0,0 +1,98 @@
+# #############################################################################
+# This Dockerfile aims to be the single source of truth for linux buildboxes on
+# all supported architectures. This is still wildly aspirational, and 
+#   a) currently only supports ARM32 & ARM64, and
+#   b) is currently only used for ARM64 builds
+# #############################################################################
+
+FROM centos:7 AS base
+
+# Automatically supplied by the Docker buildkit
+ARG TARGETARCH
+
+# #############################################################################
+# Platform-specific customisation.
+# #############################################################################
+
+FROM base AS platform-setup-arm
+ARG GOLANG_ARCH=armv6l
+ARG YUM_UPDATE_FLAGS=""
+RUN echo "armhfp" > /etc/yum/vars/basearch && \
+    echo "armv7hl" > /etc/yum/vars/arch && \
+    echo "armv7hl-redhat-linux-gnu" > /etc/rpm/platform 
+
+FROM base AS platform-setup-arm64
+ARG GOLANG_ARCH=arm64
+
+# Installing the kerenel packages causes the update to hang on aarch64, so we
+# skip upgrading them.
+ARG YUM_UPDATE_FLAGS="--exclude kernel-*"
+RUN echo "aarch64-redhat-linux-gnu" > /etc/rpm/platform 
+
+# #############################################################################
+# The minimal buildbox contains the resources required to build a minimal 
+# version of Teleport, with no optional features (e.g. BPF)
+# #############################################################################
+
+FROM platform-setup-$TARGETARCH AS minimal-buildbox
+ARG GOLANG_VERSION
+ARG RUST_VERSION
+ARG UID
+ARG GID
+
+RUN uname --all
+
+RUN (groupadd ci --gid=$GID -o && useradd ci --uid=$UID --gid=$GID --create-home --shell=/bin/sh && \
+     mkdir -p -m0700 /var/lib/teleport && chown -R ci /var/lib/teleport)
```

Comment:
This will not chmod `/var/lib/teleport` if it already exists, and if it does not already exist, then the subsequent `chown -R` is not needed. Perhaps change these two commands to:

    install --directory --mode=0700 --owner=ci --group=ci /var/lib/teleport

This could also go on its own `RUN` line, unless there is a reason to really minimise layers.

Reply:
Understood, but if `/var/lib/teleport` does exist, then the `-m0700` part of `mkdir` will be ignored, and the mode will be left unchanged, but yeah, `install` does read better.